packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -60,8 +61,21 @@ abstract class Renderer extends Object with Scheduler, Services
       );
       return true;
     });
-  }
 
+    assert(() {
+      // this service extension only works in checked mode
+      registerBoolServiceExtension(
+        name: 'repaintRainbow',
+        getter: () => debugRepaintRainbowEnabled,
+        setter: (bool value) {
+          if (debugRepaintRainbowEnabled == value)
+            return;
+          debugRepaintRainbowEnabled = value;
```

Comment:
Does this need to invalidate paint?  If it doesn't then you can even remove the if check above.